### PR TITLE
Improve task setup

### PR DESCRIPTION
**Pull request by petrdsvoboda**
I tried to complete it myself and made changes to the code and `README` based on that.

- Code
  - split components into more files, changed the file structure a bit
  - created more components (like the `HeaderAddButton`) mainly so it's easier to refer to them in the assignment and possible questions
  - changed the component API a bit, so it's more consistent
  - overall I tried to make the code's structure as easy to understand as possible, so they can focus on completing the tasks
- Readme
  - gave the tasks a name
  - tried to make the task description clearer and more descriptive?

_I've maybe gone a bit overboard, we don't have to change the assignment this much._

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,66 +6,123 @@ Please, read following instructions and let us know if anything is not clear.
...
 * Ask questions when in doubt what we expect.
-* When you are not able to solve something in reasonable amount of time, write an explanation of why
-  (what knowledge is missing, why it needs so much time, etc.), and skip it.
+* When you are not able to solve something in a reasonable amount of time, write an explanation of why
+  (what knowledge is missing, why it requires so much time, etc.), and skip it.
 * Follow all code quality principles you know and are used to.
 * You should not spend with our assignment more than 6 hours.
   When this is not doable, choose what to implement and what not to.
-* Send us your work preferably as a link to git repository (e.g. GitHub).
+* Send us your work as a link to a git repository (e.g. GitHub).
```
Comment (petrdsvoboda) [inline README.md]: Probably okay (and easier) to always request a git repo link?

File: README.md [hunk 2]
```diff
@@ -6,66 +6,123 @@ Please, read following instructions and let us know if anything is not clear.
...
+
+## Restrictions
```
Comment (petrdsvoboda) [inline README.md]: Moved them up, as I think they are important (could disqualify the task?), so it's fairer to have them first.

File: README.md [hunk 3]
```diff
@@ -6,66 +6,123 @@ Please, read following instructions and let us know if anything is not clear.
...
-- [ ] **O1**: modify the `Form` component (and related ones if needed) so the form field handles not only string, but also number and treat empty string input as `undefiened`
-- [ ] **O2**: limit amount of server calls needed to necessary minimum
 
-## Additional comments and restrictions:
-- do not modify API (props) of provided components unless achieving required tasks would not be possible otherwise,
-- do not use any component library like Material UI; do not use Tailwind,
-- feel free to do any other visual modifications that - in your opinion - makes it looking nicer,
-- feel free to install and use any public package you may need
+- [ ] **O1**: Modify the `Form` component (and other related components if needed) so the form field handles not only string values, but also numbers. Treat empty string input as `undefined`.
+- [ ] **O2**: Limit the amount of server calls needed to a necessary minimum.
```
Comment (petrdsvoboda) [inline README.md]: I am not sure how would I achieve this? Isn't this more of a requirement (or suggestion) than a task?

File: README.md [hunk 4]
```diff
@@ -6,66 +6,123 @@ Please, read following instructions and let us know if anything is not clear.
...
 ### Advanced tasks (optional)
-- [ ] **O1**: modify the `Form` component (and related ones if needed) so the form field handles not only string, but also number and treat empty string input as `undefiened`
-- [ ] **O2**: limit amount of server calls needed to necessary minimum
 
-## Additional comments and restrictions:
-- do not modify API (props) of provided components unless achieving required tasks would not be possible otherwise,
-- do not use any component library like Material UI; do not use Tailwind,
-- feel free to do any other visual modifications that - in your opinion - makes it looking nicer,
-- feel free to install and use any public package you may need
+- [ ] **O1**: Modify the `Form` component (and other related components if needed) so the form field handles not only string values, but also numbers. Treat empty string input as `undefined`.
```
Comment (petrdsvoboda) [inline README.md]: Here I was not sure, what should I do.

**Comment by fandasson** [comment]
Do you remember, why this hasn't been merged yet? @petrdsvoboda

**Comment by petrdsvoboda** [comment]
@fandasson I got carried away and possibly made the task too easy.

**Comment by fandasson** [comment]
> @fandasson I got carried away and possibly made the task too easy.

So, should't we close it then?

**Comment by petrdsvoboda** [comment]
I think I was waiting on your answers for my comments and then I planned to split this into the task description update and the code changes (only some of the ones in this PR or none at all).

File: README.md (hunk 3 above)
Comment (fandasson) [reply README.md]: If you mean O2, it is really more non-functional requirement. I think the idea was eg. not to have server call on every component re-render. But I think we can remove it from here. It typically wasn't what people had in their assignments.

File: README.md (hunk 4 above)
Comment (fandasson) [reply README.md]: Let's remove it. I'm myself not 100% sure what I meant by it

**Pull request merged**